azure-mgmt-batch/src/main/java/com/microsoft/azure/management/batch/implementation/ApplicationPackageImpl.java

```diff
@@ -53,11 +51,6 @@ public String id() {
     }
 
     @Override
-    public String name() {
```

Comment:
The original logic:
- defining a new `ApplicationPackageImpl` will set the `version` field with the user passed name (old code line 41)
- wrapping an existing `ApplicationPackageInner` into an `ApplicationPackageImpl` will copy the version field into the 1st constructor here. See https://github.com/Azure/azure-sdk-for-java/blob/master/azure-mgmt-batch/src/main/java/com/microsoft/azure/management/batch/implementation/ApplicationPackagesImpl.java#L58

In both case, the parent `name` field is always consistent with the `version` field in the inner object.